controlplane/internal/dal/dal.go

```diff
@@ -428,6 +428,34 @@ func (d *DAL) loadDeployment(ctx context.Context, deployment sql.GetLatestDeploy
 	return out, nil
 }
 
+type Metric struct {
+	RunnerKey    ulid.ULID
+	StartTime    uint64
+	EndTime      uint64
+	SourceModule string
+	SourceVerb   string
+	DestModule   string
+	DestVerb     string
+	Metric       string
+	Type         string
```

Comment:
This should be an "enum" that mirrors the SQL enum - see the `RunnerState` type in this file for an example.